File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int maxfreq = 0, ans = 0;
        unordered_map<char, int> mp;
        int j = 0;
        for(int i=0 ; i<s.length() ; i++) {
            mp[s[i]]++;
            maxfreq = max(maxfreq, mp[s[i]]);
            int len = i - j + 1;
            if(len - maxfreq <= k) {
                ans = max(ans, len);
            } else {
                mp[s[j]]--;
                j++;
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/balanced-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private boolean ans = true;

    private int fun(TreeNode root) {
        if(root == null) return 0;
        
        int lefth = 1 + fun(root.left);
        int righth = 1 + fun(root.right);

        if(Math.abs(lefth-righth) > 1) ans = false;
        return Math.max(lefth, righth);
    }

    public boolean isBalanced(TreeNode root) {
        fun(root);
        return ans;
    }
}
